src/windows/guest_list.c

```diff
@@ -179,6 +179,102 @@ void window_guest_list_open()
 }
 
 /**
+* type == 0 -> guests on ride
+* type == 1 -> guests in queue
+* type == 2 -> guests thinking about ride
+* type == 3 -> guests thinking X, opened from news item
+* ebx is number of the ride or index of the thought
+* values of eax and edx probably determine the filter name string
+*
+*  rct2: 0x006993BA
+*/
+void window_guest_list_open_with_filter(int type, int ebx)
+{
+	//RCT2_CALLPROC_X(0x006993BA, type, ebx, 0, 0, 0, 0, 0);
+
+	uint32 eax, edx, ebp;
+
+	window_guest_list_open();
+
+	_window_guest_list_selected_page = 0;
+	_window_guest_list_num_pages = 1;
+	
+	RCT2_GLOBAL(0x009AC7E0, uint8) = 0;
+	RCT2_GLOBAL(0x009AC7F0, uint8) = 0;
+
+	switch(type)
+	{
+	case 0:
+		_window_guest_list_selected_filter = 0;
+
+		ebp = ebx & 0x000000FF;
+
+		eax = RCT2_GLOBAL(0x1362942 + ebp * sizeof(rct_ride), uint16);
```

Comment:
0x13268F8 is the g_ride_list. Easy way to switch from the address to the ride offset is take the last 3 bytes in this case 0x942 and take off 0x8F8. This gives you 0x4A which is the name offset. Suggest changing this to:

```
rct_ride* ride = GET_RIDE(ebx & 0xFF);

eax = ride->name;
edx = ride->name_arguments;
```

And since a lot of this is the same for all of the cases should proabably do it before the switch.